src/cpu.ts

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { CpuProfiler, encode } from '@datadog/pprof'
+import { perftools } from '@datadog/pprof/proto/profile'
+import debug from 'debug'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ShamefulAny = any
+
+const log = debug('pyroscope::cpu')
+
+import {
+  checkConfigured,
+  config,
+  emitter,
+  SAMPLING_INTERVAL_MS,
+  SAMPLING_DURATION_MS,
+  processProfile,
+  uploadProfile,
+} from './index'
+
+const cpuProfiler = new CpuProfiler()
+
+let cpuProfilingTimer: NodeJS.Timer | undefined = undefined
+
+export function isCpuProfilingRunning(): boolean {
+  return cpuProfilingTimer !== undefined
+}
+
+export function startCpuProfiling() {
+  checkConfigured()
+
+  log('Pyroscope has started CPU Profiling')
+  cpuProfiler.start(Number(SAMPLING_INTERVAL_MS))
+
+  if (cpuProfilingTimer) {
+    log('Pyroscope has already started cpu profiling')
+    return
+  }
+
+  cpuProfilingTimer = setInterval(() => {
```

Comment:
I am trying to run, while calculating fibonacci. It seems this callback only gets called AFTER my fib code is ran. Which I believe is due to how js works: https://nodejs.org/en/docs/guides/dont-block-the-event-loop/#why-should-i-avoid-blocking-the-event-loop-and-the-worker-pool

Reply:
I mean, that's correct, the problem is that data is never flushed. Since when I call `stop` it simply cancels this setInterval, never getting the chance to upload the profile.

Reply:
stopCpuProfile not uploads last profile, should be better now